orttraining/orttraining/core/session/training_session.h

```diff
@@ -135,14 +135,32 @@ class TrainingSession : public InferenceSession {
     // If not provided, no optimizer is added.
     optional<OptimizerConfiguration> optimizer_config{};
 
+    // struct to describe a specific edge (not node_arg). This information is
+    // used in pipeline online partition tool to identify which edge to cut to
+    // make the corresponding partition.
+    struct CutEdge {
+      std::string node_arg_name;
+      optional<std::vector<std::string>> consumer_nodes;
+
+      // If the edge is unique, i.e. only have one consumer node, or all the edges
+      // with the same node_arg_name needs to be cut, specify the node_arg_name
+      // suffices.
+      CutEdge(std::string edge) : node_arg_name(edge){};
+      // If the edges with same node_arg_name belongs to different cut, i.e. some of its
+      // consumer node belongs to one partition, and some belongs to another, specify
+      // the consumer node names which you want to perform the cut on.
+      CutEdge(std::string edge, std::vector<std::string> nodes) : node_arg_name(edge), consumer_nodes(nodes){};
+    };
+    typedef std::vector<CutEdge> CutInfo;
+
     struct PipelineConfiguration {
       // If model partition happens outside ORT, this flag should be false.
       // Otherwise, use true to trigger ORT's pipeline partition.
       bool do_partition;
       // Tensors to fetch as specified by the user.
       // Each pipeline stage should pick up some strings from this field..
       std::vector<std::string> fetch_names;
-      // [TODO] Add cut information.
+      std::vector<CutInfo> cut_list;
```

Comment:
>std::vector<CutInfo> cut_list; [](start = 6, length = 30)

What is this? cut_list[i] contains the edges between the first and the second pipeline stages? For vector fields, we need to put a definition for its elements.

Reply:
cut_list[i] contains the CutInfo to make the partition between stage i and stage i+1. Comment updated.

---
In reply to: [429521450](https://github.com/microsoft/onnxruntime/pull/3996#discussion_r429521450) [](ancestors = 429521450)